### DISCUSSION TITLE
Tried to add a new Driver following document but says it can't load the .py

### DISCUSSION THREAD

**Discussion by johnw230873** [tuya/tuya-home-assistant#364] [Q&A]
I have a Dual-Network Secuirty Alarm.
Looking at the output I know that the status of "master_mode" is the value I need to change.

**Not set** 
.....
"status": [
{
"code": "master_mode",
"value": "disarmed"
},
......

**Set** 
.....
"status": [
{
"code": "master_mode",
"value": "arm"
},
.....

Know this I first thought I would use the switch.py adding the TUYA_SUPPORT_TYPE of **mal** and then add an extra dp_code.

Reading more of the wiki page "How to Develop a New Driver" I thought I would add an extra PY call alarm.py and add the extra code into that file. From reading the file all I expected I needed to do was add a "alarm.py" in **const.py** in the **TUYA_SUPPORT_HA_TYPE** hence 

TUYA_SUPPORT_HA_TYPE = [
    "switch",   
    "fan",
    "cover",
    "climate",
    "light",
    "sensor",
    "binary_sensor",
    "humidifier",
    "number",
    "vacuum",
   "alarm"
]

but when I do this I get an error when Home Assistant is reloaded saying it can't local alarm.py. I have checked the logs and they don't indicate why.

I also noticed that the py of alarm_control_panel.py was in the TUYA_SUPPORT_HA_TYPE, does this get loaded.

I would love to work with someone to get this going but stuck as the wiki isn't working as expected.

So current issue
why can't I create an alarm.py
is alarm_control_panel.py getting loaded even though it isn't in TUYA_SUPPORT_HA_TYPE

If I know there are only 2 modes for the alarm should I be using switch.py

thanks

**Comment by tsutsuku** [comment]
1. I think you should use alarm_control_panel.py as your device is a Security Alarm.
2. You can't create an alaram.py because the entity file must be the components naming, like alarm_control_panel.py for AlarmControlPanelEntity in components/alarm_control_panel, switch.py for SwitchEntity in components/switch.
3. When we develop PIR driver, we used alarm_control_panel first, but then we found it's unsuitable for PIR, so we change it to binary sensor and disable alarm_control_panel for future driver development without deleting it.

Welcome to create a PR when you finish your driver.

**Reply by johnw230873** [reply]
Thanks tsutsuka
I am finding it harder than expected to get these functions working as very new to Phyton but I have altered the line **from homeassistant.const** to also include the states  STATE_ALARM_DISARMED, STATE_ALARM_ARMED_AWAY

I've then added in **mal** to the **TUYA_SUPPORT_TYPE** 
**DPCODE_MASTER_MODE = "master_mode"**

also added this into the **_setup_entities**

> if DPCODE_MASTER_MODE in device.status:
>   entities.append( 
>     TuyaHaAlarm( 
>               device, 
>               device_manager,
>               (
>                         lambda d: STATE_ALARM_DISARMED
>                         if d.status.get(DPCODE_MASTER_MODE, 1) == "disarmed"
>                         else STATE_ALARM_ARMED_AWAY
>                     ),
>                 )   
>             )
> 


At this stage I can now see the entity in the Home Assistant portal and it changes status when the Alarm is armed or disabled.
![image](https://user-images.githubusercontent.com/3125664/129982756-ed9c87cf-e20d-4051-a2c8-fa16fdbc906a.png)

What I still can't seem to do it make the buttons work on the Alarm panel, is there some advise you could provide, such as use another .py to template off knowing that the states are the simple true/false?

**Reply by tsutsuku** [reply]
Can you provide your alarm device info and device specifications from the logs?

**Reply by johnw230873** [reply]
I've got a bit further but this has me stuffed.
the device now reports all master_mode values and the GUI on the lovelace portal looks fine.
If the alarm is disarmed I can't "Arm away" or "Arm Home", however if it is armed then I can "disarmed" also if I change the command in the "Alarm_disarm" def to say "home" this works as well, so I command is correct.

I have also checked events under developer tools and I can HA is firing events alarm_arm_home and alarm_arm_away so unsure why they are being sent to Tuya.

Here is the latest .py

[alarm_control_panel.py.txt](https://github.com/tuya/tuya-home-assistant/files/7013483/alarm_control_panel.py.txt)

[device info.txt](https://github.com/tuya/tuya-home-assistant/files/7013486/device.info.txt)
